.github/workflows/build-pulse.yml

```diff
@@ -0,0 +1,60 @@
+name: BuildPulse
+
+on:
+  workflow_run:
+    workflows: ["Test"]
+    types:
+      - completed
+  workflow_dispatch:
+    inputs:
+      artifacts_url:
+        description: The URL of the artifacts for the Test workflow run
+        required: true
+      head_commit_id:
+        description: The commit ID of the head commit of the Test workflow run
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  actions: read
+
+jobs:
+  buildpulse:
+    name: Upload JUnit XML report to BuildPulse
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion == 'success' || github.event.workflow_run.conclusion == 'failure'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.head_commit_id || github.event.workflow_run.head_commit.id }}
+      - id: reports
+        run: mktemp -d | xargs -0 -I{} echo "path={}" | tee -a $GITHUB_OUTPUT
+      - name: Download all JUnit XML reports from the Test workflow run
+        env:
+          GH_TOKEN: ${{ github.token }}
+          artifacts_url: ${{ github.event.inputs.artifacts_url || github.event.workflow_run.artifacts_url }}
+          reports_path: ${{ steps.reports.outputs.path }}
+        run: |
+          pushd $reports_path
+          while read -r archive_download_url; do
+            if [[ -z "$archive_download_url" ]]; then
+              continue
+            fi
+            gh api "$archive_download_url" > archive.zip
+            unzip archive.zip
+            rm archive.zip
+          done <<< "$(gh api "$artifacts_url" | jq -r '.artifacts | map(.archive_download_url) | .[0]')"
```

Comment:
Eek.  Is this really the cleanest way to get the download URL?  I'm surprised.
I also don't love the shell scripting here that the input for the while loop is defined at the end.

Reply:
Replaced with actions/download-artifacts. Apparently it supports cross-run downloads now.